### PR TITLE
FacetGrid docstrings

### PR DESCRIPTION
**Pull request by zmoon**
- [x] Passes `pre-commit run --all-files`
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

Summary:
* Add `FacetGrid.add_quiverkey` to `api.rst`
* Other edits similar to #5285

### CODE REVIEW

**Comment by max-sixty** [comment]
Thanks @zmoon ! I'm going to politely ask @keewis to review this since he has driven many of our improvements in this area.

File: xarray/plot/facetgrid.py [hunk 1]
```diff
@@ -110,7 +111,8 @@ def __init__(
         size : scalar, optional
             Height (in inches) of each facet. See also: ``aspect``
         subplot_kws : dict, optional
-            Dictionary of keyword arguments for matplotlib subplots
+            Dictionary of keyword arguments for Matplotlib subplots
+            (:func:`matplotlib.pyplot.subplots`).
```
Comment (keewis) [inline xarray/plot/facetgrid.py:115]: I would use `:py:func:` here:
```suggestion
            (:py:func:`matplotlib.pyplot.subplots`).
```

**Review by keewis — APPROVED** [review]
thanks a lot, @zmoon. I only have one minor suggestion.

File: xarray/plot/facetgrid.py (hunk 1 above)
Comment (zmoon) [reply xarray/plot/facetgrid.py]: Thanks @keewis, I can make that change in #5285 as well.

**Comment by dcherian** [comment]
Thanks @zmoon. This will really help our users.

**Comment by zmoon** [comment]
@dcherian in some sense this is a "user visible change", but quite minor. I wasn't sure if I should have added to *What's New* or not.

**Pull request merged**